geode-connectors/src/main/java/org/apache/geode/connectors/jdbc/internal/cli/DestroyMappingCommand.java

```diff
@@ -63,22 +71,101 @@ public ResultModel destroyMapping(@CliOption(key = DESTROY_MAPPING__REGION_NAME,
 
   @Override
   public boolean updateConfigForGroup(String group, CacheConfig cacheConfig, Object configObject) {
-    String region = (String) configObject;
-    RegionMapping existingCacheElement = cacheConfig.findCustomRegionElement("/" + region,
-        RegionMapping.ELEMENT_ID, RegionMapping.class);
-
-    if (existingCacheElement != null) {
-      cacheConfig
-          .getRegions()
-          .stream()
-          .filter(regionConfig -> regionConfig.getName().equals(region))
-          .forEach(
-              regionConfig -> regionConfig.getCustomRegionElements().remove(existingCacheElement));
+    String regionName = (String) configObject;
+    RegionConfig regionConfig = findRegionConfig(cacheConfig, regionName);
+    if (regionConfig == null) {
+      return false;
+    }
+    boolean modified = false;
+    modified |= removeJdbcMappingFromRegion(regionConfig);
+    modified |= removeJdbcQueueFromCache(cacheConfig, regionName);
+    RegionAttributesType attributes = getRegionAttributes(regionConfig);
+    modified |= removeJdbcLoader(attributes);
+    modified |= removeJdbcWriter(attributes);
+    modified |= removeJdbcAsyncEventQueueId(attributes, regionName);
+    return modified;
+  }
+
+  private boolean removeJdbcLoader(RegionAttributesType attributes) {
+    DeclarableType cacheLoader = attributes.getCacheLoader();
+    if (cacheLoader != null) {
+      if (JdbcLoader.class.getName().equals(cacheLoader.getClassName())) {
+        attributes.setCacheLoader(null);
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private boolean removeJdbcWriter(RegionAttributesType attributes) {
+    DeclarableType cacheWriter = attributes.getCacheWriter();
+    if (cacheWriter != null) {
+      if (JdbcWriter.class.getName().equals(cacheWriter.getClassName())) {
+        attributes.setCacheWriter(null);
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private boolean removeJdbcAsyncEventQueueId(RegionAttributesType attributes, String regionName) {
+    String queueName = CreateMappingCommand.createAsyncEventQueueName(regionName);
+    String queueIds = attributes.getAsyncEventQueueIds();
+    if (queueIds == null) {
+      return false;
+    }
+    List<String> queues = new ArrayList<>(Arrays.asList(queueIds.split(",")));
+    if (queues.contains(queueName)) {
+      queues.remove(queueName);
+      String newQueueIds = String.join(",", queues);
```

Comment:
It looks like we will be setting empty string ("") as asyncId, if the queue becomes empty after removal, do we need a check for empty queue here.

Reply:
In that case we set this attribute to the empty string which means it has no asyncQueueIds. We have a unit test that confirms this. So I don't see any need to check for an empty queue here. The String.join call handles it correctly.